packages/go/analysis/ad/ntlm.go

```diff
@@ -245,44 +249,48 @@ func GetCoerceAndRelayNTLMtoADCSEdgeComposition(ctx context.Context, db graph.Da
 		return nil, err
 	}
 
-	if err := traversalInst.BreadthFirst(ctx, traversal.Plan{
-		Root: endNode,
-		Driver: coerceAndRelayNTLMtoADCSPath1Pattern(domainNode.ID).Do(func(terminal *graph.PathSegment) error {
-			var enterpriseCANode *graph.Node
-			terminal.WalkReverse(func(nextSegment *graph.PathSegment) bool {
-				if nextSegment.Node.Kinds.ContainsOneOf(ad.EnterpriseCA) {
-					enterpriseCANode = nextSegment.Node
-				}
-				return true
-			})
+	for _, n := range startNodes.Slice() {
+		if err := traversalInst.BreadthFirst(ctx, traversal.Plan{
+			Root: n,
+			Driver: coerceAndRelayNTLMtoADCSPath1Pattern(domainNode.ID).Do(func(terminal *graph.PathSegment) error {
+				var enterpriseCANode *graph.Node
+				terminal.WalkReverse(func(nextSegment *graph.PathSegment) bool {
+					if nextSegment.Node.Kinds.ContainsOneOf(ad.EnterpriseCA) {
+						enterpriseCANode = nextSegment.Node
+					}
+					return true
+				})
 
-			lock.Lock()
-			candidateSegments[enterpriseCANode.ID] = append(candidateSegments[enterpriseCANode.ID], terminal)
-			path1EnterpriseCAs.Add(enterpriseCANode.ID.Uint64())
-			lock.Unlock()
+				lock.Lock()
+				candidateSegments[enterpriseCANode.ID] = append(candidateSegments[enterpriseCANode.ID], terminal)
+				path1EnterpriseCAs.Add(enterpriseCANode.ID.Uint64())
+				lock.Unlock()
 
-			return nil
-		}),
-	}); err != nil {
-		return nil, err
+				return nil
+			}),
+		}); err != nil {
+			return nil, err
+		}
 	}
 
-	if err := traversalInst.BreadthFirst(ctx, traversal.Plan{
-		Root: endNode,
-		Driver: coerceAndRelayNTLMtoADCSPath2Pattern(domainNode.ID, path1EnterpriseCAs).Do(func(terminal *graph.PathSegment) error {
-			enterpriseCANode := terminal.Search(func(nextSegment *graph.PathSegment) bool {
-				return nextSegment.Node.Kinds.ContainsOneOf(ad.EnterpriseCA)
-			})
+	for _, n := range startNodes.Slice() {
```

Comment:
Can this be done in the same `for` above?

Reply:
No because we need to build the full enterpriseCA list from part 1 to join on first